src/mlagility/api/cloud.py

```diff
@@ -365,6 +396,82 @@ def execute_gpu_remotely(
     sys.stdout = sys.stdout.terminal
 
 
+def execute_gpu_locally(
+    state: build.State, log_execute_path: str, iterations: int
+) -> None:
+    """
+    Execute Model on the local GPU
+    """
+
+    # Redirect all stdout to log_file
+    sys.stdout = build.Logger(log_execute_path)
+
+    # Setup local execution folders to save outputs/ errors
+    username = getpass.getuser()
+    output_dir = f"/home/{username}/mlagility_local_cache"
```

Comment:
How is this functionally different from the `mlagility cache` at ~/.cache/mlagility?

Reply:
This is for symmetry with our remote testing. On the remote tests we create a scratchpad folder `mlagility_remote_cache` and on the local tests we create a scratchpad folder `mlagility_local_cache`. But this folder gets deleted once the test is complete.

Reply:
Created #101